pkg/lang/capabilities.go

```diff
@@ -52,64 +57,87 @@ func CompositePreprocessor(preprocessors ...CommentPreprocessor) CommentPreproce
 	}
 }
 
+func IsCLineCommentBlock(previous, current *sitter.Node) bool {
+	return previous != nil && current != nil &&
+		current.StartPoint().Row-previous.StartPoint().Row == 1 &&
+		strings.HasPrefix(current.Content(), "//") &&
+		strings.HasPrefix(previous.Content(), "//")
+}
+
+func IsHashCommentBlock(previous, current *sitter.Node) bool {
+	return previous != nil && current != nil &&
+		current.StartPoint().Row-previous.StartPoint().Row == 1 &&
+		strings.HasPrefix(current.Content(), "#") &&
+		strings.HasPrefix(previous.Content(), "#")
+}
+
 // NewCapabilityFinder creates a struct that you can use to find capabilities (annotations) within a source file.
 //
 // To do this, you provide a `sitterQuery` that looks for comments nodes that contain the `klotho::` annotations,
 // as well as a preprocessor that runs over each of those nodes.
 //
 // If a source file contains multiple comment nodes in a row (as identified by having equal `.Type()`s), those comments
 // will be preprocessed individually, but then merged into a single annotation.
-func NewCapabilityFinder(sitterQuery string, preprocessor CommentPreprocessor) core.CapabilityFinder {
+func NewCapabilityFinder(sitterQuery string, preprocessor CommentPreprocessor, mergePredicate MergeCommentsPredicate) core.CapabilityFinder {
 	return &capabilityFinder{
-		sitterQuery:  sitterQuery,
-		preprocessor: preprocessor,
+		sitterQuery:            sitterQuery,
+		preprocessor:           preprocessor,
+		mergeCommentsPredicate: mergePredicate,
 	}
 }
 
-// FindAllCapabilities finds all of the annotations (ie, capabilities) in a SourceFile.
+// FindAllCapabilities finds all the annotations (ie, capabilities) in a SourceFile.
 func (c *capabilityFinder) FindAllCapabilities(f *core.SourceFile) (core.AnnotationMap, error) {
 	var merr multierr.Error
 	capabilities := make(core.AnnotationMap)
-	for _, block := range c.findAllCommentsBlocks(f) {
-		cap, err := annotation.ParseCapability(block.comment)
-		if cap == nil {
-			continue
-		}
-		annotation := &core.Annotation{Capability: cap, Node: block.node}
-		if err != nil {
-			merr.Append(core.NewCompilerError(f, annotation, errors.Wrap(err, "error parsing annotation")))
-			continue
+	for _, block := range c.findAllCommentBlocks(f) {
+		caps, err := annotation.ParseCapabilities(block.comment)
```

Comment:
If this returns no caps, then the err is unchecked. Getting 1 error per cap when it's only generated once is odd.

Reply:
Yeah, the error handling here needs to be updated to account for multiple annotations in a block.